### PR TITLE
Closes #21, Delete all

### LINKED ISSUE #21: Feature request: delete-all command
Same as `get-all` but for recursive deletion:
`vault -U https://myvault.test:8200 -b myapp -w delete-all db/`

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes #21

Based on #48, rebase after merge.

### CODE REVIEW

**Comment by ewjoachim** [comment]
Aw gosh tests fail.

**Comment by ewjoachim** [comment]
Tests fail no more.

File: vault_cli/client.py [hunk 1]
```diff
@@ -217,6 +237,16 @@ def _merge_secrets(self, secrets: types.JSONDict) -> types.JSONDict:
...
         return secrets
 
+    def delete_all_secrets(self, *paths: str) -> Iterable[str]:
+        """
+        Recursively deletes all the secrets at the given paths.
+        """
+        for path in paths:
+            secrets_paths = self._browse_recursive_secrets(path=path)
+            for secret_path in secrets_paths:
+                yield secret_path
```
Comment (mgu) [inline vault_cli/client.py]: it's weird that this method `yield` the item before deleting item :)

Comment (ewjoachim) [reply vault_cli/client.py]: Yes, to give a chance for confirming your choice before it’s too late. Is it that much of a strange pattern ?

Comment (mgu) [reply vault_cli/client.py]: the goal is clear but the name of the function is misleading :)

Comment (ewjoachim) [reply vault_cli/client.py]: Any suggestion ?

Comment (mgu) [reply vault_cli/client.py]: `yield_and_delete_all_secrets` :D

Comment (jacquesrott) [reply vault_cli/client.py]: `pop_all_secrets` ? :)

Comment (ewjoachim) [reply vault_cli/client.py]: Je vais merger et on améliorera ça après

**Pull request merged**